### PR TITLE
Flutter iOS Interactive Keyboard: Fixing Animation Issue

### PR DESCRIPTION
**Pull request by Matt2D**
This PR addresses an issue with the animation of the keyboard. In iOS 16.0 a delay was included in UIView becomeFirstResponder where the areAnimationsEnabled boolean is no longer immediately read. In response to this issue a delay is added that allows for the animation to be properly disabled. 

Design Document:
https://docs.google.com/document/d/1-T7_0mSkXzPaWxveeypIzzzAdyo-EEuP5V84161foL4/edit?pli=1

Issues Address:
https://github.com/flutter/flutter/issues/57609

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm [hunk 1]
```diff
@@ -2369,8 +2369,12 @@ - (void)dismissKeyboardScreenshot {
 - (void)showKeyboardAndRemoveScreenshot {
   [UIView setAnimationsEnabled:NO];
   [_cachedFirstResponder becomeFirstResponder];
-  [UIView setAnimationsEnabled:YES];
-  [self dismissKeyboardScreenshot];
+  // UIKit does not immediately access the areAnimationsEnabled Boolean so a delay is needed before
+  // returned
+  dispatch_after(dispatch_time(DISPATCH_TIME_NOW, 0.1 * NSEC_PER_SEC), dispatch_get_main_queue(), ^{
```
Comment (cbracken) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm:2374]: We should extract a constant and use that here:
```objc
// Detailed explanation of why this is necessary.
constexpr kKeyboardAnimationDelay = 0.1 * NSEC_PER_SEC;
```

Comment (Matt2D) [reply shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: Definitely! Just added.

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm [hunk 2]
```diff
@@ -21,6 +21,10 @@
 // it is activated.
 static constexpr double kUITextInputAccessibilityEnablingDelaySeconds = 0.5;
 
+// A delay before reenabling the UIView areAnimationsEnabled to YES
+// in order for becomeFirstResponder to recieve the proper value
```
Comment (Logharaa) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm:25]:
```suggestion
// in order for becomeFirstResponder to receive the proper value
```

**Review by hellohuanlin — APPROVED** [review]
LGTM!

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm (hunk 2 above)
Comment (Matt2D) [reply shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: Thank you! Those will slip by me from time to time.

**Review by cbracken — COMMENTED** [review]
woohoo - sorry; meant to hit lgtm on my last review but posted as a comment instead. LGTM!

**Pull request merged**